tests/ext-popup.test.js

```diff
@@ -0,0 +1,246 @@
+
+  const promiseErrorChanged2 = observeChange(errorText);
+  stopBtn.click();
+  await promiseErrorChanged2;
+
+  expect(errorText.textContent).toBe("Monitoring couldn't be stopped");
+
+  expect(startBtn.getAttribute('disabled')).toEqual('disabled');
+  expect(stopBtn.getAttribute('disabled')).toBeNull();
+
+  expect(monitorStatus.textContent).toBe('Extensions are being monitored');
+  expect(monitorStatus.classList.contains('failure')).toBeFalsy();
+  expect(monitorStatus.classList.contains('success')).toBeTruthy();
+});
+
+test('wrong parameter is passed to handleEvent', async () => {
+  document.body.innerHTML = popupBody;
+
+  const getMonitorStatusFn = jest.spyOn(ExtListen, 'getMonitorStatus');
+
+  const popup = new Popup();
+  // initially there will be no monitored extensions
+  getMonitorStatusFn.mockResolvedValue(false);
+
+  await popup.init();
+
+  const target = { id: 'invalid' };
+  const invalidType = 'invalid';
+
+  let thrownError = popup.handleEvent({
+    type: 'click',
+    target,
+  });
+  await expect(thrownError).rejects.toThrowError(
+    'wrong event target found ' + target
+  );
+
+  await popup.init();
+
+  thrownError = popup.handleEvent({
+    type: invalidType,
+    target: { id: 'invalid' },
+  });
+  await expect(thrownError).rejects.toThrowError(
+    'wrong event type found ' + invalidType
+  );
+});
+
+test('clicking "view activity logs page" should open the activitylog extension page', async () => {
+  document.body.innerHTML = popupBody;
+
+  const getMonitorStatusFn = jest.spyOn(ExtListen, 'getMonitorStatus');
+  const openActivityLogPageFn = jest.spyOn(ExtListen, 'openActivityLogPage');
+
+  openActivityLogPageFn.mockResolvedValue();
+
+  const popup = new Popup();
+
+  const startBtn = popup.startMonitorAllBtn;
+  const stopBtn = popup.stopMonitorAllBtn;
+  const viewExtPageBtn = popup.viewActivityLogBtn;
+  const monitorStatus = popup.monitorStatusText;
+  const errorText = popup.errorMsgText;
+
+  // initially there will be no monitored extensions
+  getMonitorStatusFn.mockResolvedValue(false);
+
+  await popup.init();
+
+  expect(stopBtn.getAttribute('disabled')).toEqual('disabled');
+  expect(startBtn.getAttribute('disabled')).toBeNull();
+
+  expect(monitorStatus.textContent).toBe('No extensions are being monitored');
+  expect(monitorStatus.classList.contains('failure')).toBeTruthy();
+  expect(monitorStatus.classList.contains('success')).toBeFalsy();
+  expect(errorText.textContent).toBe('');
+
+  viewExtPageBtn.click();
+
+  expect(openActivityLogPageFn).toHaveBeenCalled();
+
+  // nothing changes from previous check
+  expect(stopBtn.getAttribute('disabled')).toEqual('disabled');
+  expect(startBtn.getAttribute('disabled')).toBeNull();
+
+  expect(monitorStatus.textContent).toBe('No extensions are being monitored');
+  expect(monitorStatus.classList.contains('failure')).toBeTruthy();
+  expect(monitorStatus.classList.contains('success')).toBeFalsy();
+  expect(errorText.textContent).toBe('');
```

Comment:
This is a very long test to only test a small number of lines :-)

We could rip it out or at least try to make it smaller, e.g. all this part is checking something that is already being tested in another test case, and in this test we are only interested in checking that the "view logs" button is going to open the extension page, and so we could reduce it a lot by just doing and checking the bare minimum, e.g.
```
  const viewExtPageBtn = popup.viewActivityLogBtn;
  getMonitorStatusFn.mockResolvedValue(false);
  await popup.init();
  viewExtPageBtn.click();
  expect(openActivityLogPageFn).toHaveBeenCalled();
```

Besides that, (in general not just related to this particular test case) don't worry too much about getting the full coverage for now, we are not going to block the PR until the coverage is 100%, as long as the code is nice and clean and does what it should in the most simple way possible we are going to merge the PR and move forward.

Writing these unit tests is currently meant to help you to:

- learn more about testing (which is always useful)
- re-look to your code from other perspectives, find issues that you may have missed (and eventually test explicitly the part that matter the most)

Reply:
Thanks for explaining :-)